internal/assert/assertions.go

```diff
@@ -0,0 +1,92 @@
+package assert
+
+// Testing check shortcuts
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+// Log message and fail if actualValue doesn't match expectedValue
+// *Note that DeepEqual will check the value pointed to by two pointers, ie don't use for checking
+//	pointer equality.
+func Equal(t *testing.T, expectedValue interface{}, actualValue interface{}, cm ...interface{}) bool {
+	if !reflect.DeepEqual(expectedValue, actualValue) {
+		msg := fmt.Sprintf("%v != %v (expected)", actualValue, expectedValue)
+		if len(cm) > 0 {
+			msg = msg + " - " + fmt.Sprintf(cm[0].(string), cm[1:]...)
+		}
+		return Fail(t, msg)
+	}
+	return true
+}
+
+// Log message and fail if testValue is not true.
+func True(t *testing.T, testValue bool, cm ...interface{}) bool {
+	if !testValue {
+		msg := fmt.Sprintf("%v is false", testValue)
+		if len(cm) > 0 {
+			msg = msg + " - " + fmt.Sprintf(cm[0].(string), cm[1:]...)
+		}
+		return Fail(t, msg)
+	}
+	return true
+}
+
+// Log message and fail if testValue is not nil.
+func Nil(t *testing.T, testValue interface{}, cm ...interface{}) bool {
+	if !isNil(testValue) {
+		msg := fmt.Sprintf("%v (%T) is not nil", testValue, testValue)
+		if len(cm) > 0 {
+			msg = msg + " - " + fmt.Sprintf(cm[0].(string), cm[1:]...)
+		}
+		return Fail(t, msg)
+	}
+	return true
+}
+
+// Log message and fail if testValue is nil.
+func NotNil(t *testing.T, testValue interface{}, cm ...interface{}) bool {
+	if isNil(testValue) {
+		msg := fmt.Sprintf("%v is nil", testValue)
+		if len(cm) > 0 {
+			msg = msg + " - " + fmt.Sprintf(cm[0].(string), cm[1:]...)
+		}
+		return Fail(t, msg)
+	}
+	return true
+}
+
+// Fail performs t.Errorf(), but modifies the message with the file and line of the caller.
+func Fail(t *testing.T, failureMessage string) bool {
+	//	content := [][2]string{
+	//		{"Error Trace", strings.Join(CallerInfo(), "\n\r\t\t\t")},
+	//		{"Error", failureMessage},
+	//	}
+	_, file, line, _ := runtime.Caller(2)
```

Comment:
is an error being ignored here? Maybe throw a panic on error if you don't want to return it in this instance?